### PR TITLE
Reverts both "[#4245] Allow password to nil (#4261)" and "Add more tests (#4970)"

### LINKED ISSUE #5033: v4.6.0 breaks activeadmin forms that have `password` field
## Environment

- Ruby 2.6.1
- Rails 5.2.2
- Devise 4.6.1

## Current behavior

It appears that PR #4261 introduced a change where `encrypted_password` could be set to `nil` even when no attempt was made to change `password`.  The relevant comment is here https://github.com/plataformatec/devise/pull/4261/files#r77820245

This results in user model forms that have a password field in activeadmin to break with a validation failure when the password field remains empty on submit:

```
ActiveRecord::NotNullViolation (PG::NotNullViolation: ERROR:  null value in column "encrypted_password" violates not-null constraint
```

## Expected behavior

No attempt should be made to set `encrypted_password` to `nil` if the `password` field doesn't have a value.  This was the case in devise versions prior to v4.6.0.

### PR DESCRIPTION
**Pull request by mracos**
Closes #5033 

More info in https://github.com/plataformatec/devise/issues/5033#issuecomment-476386275.

### CODE REVIEW

File: lib/devise/models/database_authenticatable.rb [hunk 1]
```diff
@@ -60,17 +60,18 @@ def self.required_fields(klass)
       # the hashed password.
       def password=(new_password)
         @password = new_password
-        self.encrypted_password = password_digest(@password) 
+        self.encrypted_password = password_digest(@password) if @password.present?
       end
 
       # Verifies whether a password (ie from sign in) is the user password.
       def valid_password?(password)
+        return false if password.blank?
```
Comment (tegon) [inline lib/devise/models/database_authenticatable.rb]: I guess this line doesn't need to be here, right? It was added in https://github.com/plataformatec/devise/pull/4261/files#diff-19b4dd928714c72f1338874351e8ff2dR45

Comment (mracos) [reply lib/devise/models/database_authenticatable.rb]: Fixed by doing the reverts on chronological order

**Comment by jbeker** [comment]
Thanks! Will there be a release shortly that includes this?

**Comment by mracos** [comment]
@jbeker We already released 4.6.2 including these changes 😄

https://github.com/plataformatec/devise/releases/tag/v4.6.2
https://rubygems.org/gems/devise/versions/4.6.2

**Comment by jbeker** [comment]
@mracos brilliant! thanks!

**Pull request merged**